lib/lightning_web/components/new_inputs.ex

```diff
@@ -17,13 +17,23 @@ defmodule LightningWeb.Components.NewInputs do
   """
   attr :id, :string, default: ""
   attr :type, :string, default: "button", values: ["button", "submit"]
-  attr :class, :any, default: nil
+  attr :class, :any, default: ""
   attr :rest, :global, include: ~w(disabled form name value)
   attr :tooltip, :any, default: nil
+  attr :cancel, :boolean, default: false
 
   slot :inner_block, required: true
 
   def button(assigns) do
+    custom_classes =
+      if assigns[:cancel] do
+        "bg-neutral-100 hover:bg-neutral-300 text-neutral-950"
+      else
+        "bg-primary-600 hover:bg-primary-700 text-white"
+      end
+
+    assign(assigns, :class, fn class -> class ++ custom_classes end)
```

Comment:
Just a tip here, since LiveView can interpolate lists into attrs. Instead of creating a new assign you can add to the class attr below with `..., @custom_classes, @class`.

Reply:
Yes, thanks! We use on other parts of the code ... here it concatenates as the custom classes are defined internally (have opted for the smallest change but could avoid string concatenation though adding a new `@custom_classes` assign to put it as an element of the class list)

Reply:
I've updated the code to use a different `assign` called `visual_class`. This assign will be responsible of holding all styles related to `color` and so it can be swapped by a different set entirely.
I'm not sure if `visual_class` is the right name for it though. Maybe I should call it `color_class` instead 🤔 ?